### PR TITLE
fix(materials): ground revision-gate coverage in shipped resume text

### PR DESCRIPTION
**Pull request by ebarti**
## What changed

The post-generation revision gate counted a requirement as covered whenever the model's self-declared `GeneratedClaimMapping`s asserted it — the shipped text was never consulted. The requirement-led audit chips separately bound requirements to bullets by verbatim keyword matching. The same accepted artifact could therefore persist **"Must-have coverage: 100% · passed"** beside a provenance audit showing **"4/9 requirements covered"** with must-haves flagged missing from the tailored resume (observed live on the Apply Review surface for the Digital Hub Director job, generation 32).

This PR makes **claim grounding** the single coverage criterion, owned by a new pure domain module `claim_grounding.py`:

- A coverage-bearing claim counts only when bound to a **shipped rendered line**: location resolution (`experience.{id}.bullets[i]` → `experience:{id}#{i}`, all payload-surface aliases) plus token-sequence text binding; a text-scan fallback covers location drift.
- A **voice-reworded** bullet honors its pre-voice text (voice keeps bullet identity 1:1 and re-runs the fabrication gate), so meaning-preserving rewording never unships a claim.
- The assembler's **policy-gated summary swap** (`allow_summary_rewrite: no`) makes a proposed-but-unshipped rewrite honestly ungrounded.
- `score_generated_resume_against_target` now requires the grounding; requirements asserted only by unshipped claims are reported as `claimed_only_requirement_ids`, count as **uncovered**, and their prioritized fixes say the claim text never shipped — feeding the existing revision loop so it repairs real gaps (self-healing) instead of echoing self-assessment.
- The per-candidate gate renders the candidate's shipped lines (pure, assembler-mirroring `build_bullet_provenance`) and records an inspectable `grounding` audit beside `fit_score`/`revision_decision`.
- Accepted artifacts persist provenance rows **enriched with grounded claim requirement/evidence links**, so the audit chips and the "N/M requirements covered" badge derive from the same criterion as the gate — the three surfaces can no longer disagree for new generations.
- The shipped (post-voice) artifact persists a lifecycle-labeled `post_generation_fit_final` record (`lifecycle: post_voice_shipped`, gate thresholds, explicit residual warnings when voice degraded grounded coverage after acceptance).

## Why

Root-cause fix for the Apply Review coverage contradiction. Per the repo's auditability discipline: the gate value shown on the approve surface must be measured against the generated artifact, not the model's word; post-acceptance signals must be lifecycle-labeled; and the displayed claim sources must converge.

## How it was validated

- `uv --project workers/automation run --extra dev pytest -q` → **1666 passed** (was 1661; 11 new grounding tests, gate tests rewritten to grounded semantics, all voice/provenance/retailor integration tests green unchanged in behavior except the intended gate semantics).
- `test_grounded_gate_reproduces_apply_review_contradiction_shape` reproduces the exact generation-32 state (claims assert 9/9, one shipped summary line carries r3–r6) and proves the gate now reports 50% must-have coverage, fails, and routes a revision with shipped-resume fix instructions.
- `uv --project workers/automation run --extra dev ruff check .` → clean on changed files.
- Docs: new High-Risk Regression Areas row in `docs/local-reliability-qa.md`.

Stacked follow-up (PR2) will label the gate basis + lifecycle on the API/web read side and fix the "Attempt 2/1" display semantics.

### CODE REVIEW

**Review by ebarti — COMMENTED** [review]
PR Reviewer - pr-reviewer: automated read-only review of the grounded coverage gate. Full suite green (1666 passed) and ruff clean. Gate: FAIL on one High finding (claim `evidence_ids` enrichment leaks into the #216 keyword-coverage audit and re-credits keywords as covered off unrelated evidence). One Medium (missing use_cases-level tests / over-claimed docs coverage) and one Low (empty-rows gate record on coverage-graph-only errors) follow inline.

File: workers/automation/src/jobhunter/domain/materials/claim_grounding.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+        if not claim_requirements and not claim_evidence:
+            out.append(row)
+            continue
+        out.append(
+            replace(
+                row,
+                requirement_ids=tuple(
+                    dict.fromkeys((*claim_requirements, *row.requirement_ids))
+                ),
+                evidence_ids=tuple(dict.fromkeys((*row.evidence_ids, *claim_evidence))),
```
Comment (ebarti) [inline workers/automation/src/jobhunter/domain/materials/claim_grounding.py]: PR Reviewer - pr-reviewer

Severity: High
Gate: Fail
Evidence: `enrich_provenance_requirements` unions the grounded claim's `evidence_ids` onto the carrying provenance row (claim_grounding.py:273). `_voice_and_audit` then calls `_coverage_for` on these enriched rows on every return path (use_cases.py:2925/2939/2961/2973), and `compute_keyword_coverage` credits a planned keyword as `covered` whenever `row.evidence_ids` is truthy (coverage_audit.py:212).
Reproduction: Reproduced with a standalone script against compute_keyword_coverage: a bullet "Led Kubernetes platform migration." with empty builder evidence + a planned keyword "kubernetes" that traces to no corpus is `missing`; after a coverage claim carrying an UNRELATED evidence id (ev_python) grounds to it, the enriched row gains ('ev_python',) and "kubernetes" flips to `covered` (BEFORE missing=('kubernetes',) -> AFTER covered=('kubernetes',)).

This regresses the #216 invariant (merged same day, c04f0528): "A requirement FK is deliberately NOT sufficient ... crediting coverage off that binding would let a keyword ground itself and reward the stuffing this audit exists to catch." Model-declared claim evidence is exactly the self-declared binding #216 stopped trusting, and the coarse `row.evidence_ids` truthy check credits ALL keywords in the bullet, not only those the claim's evidence supports. The corrupted covered/missing split is persisted in `keyword_coverage_v2` and rendered on Apply Review, and no test guards it. docs/local-reliability-qa.md explicitly lists "keyword coverage ... silently credited as covered without evidence" as a must-prevent regression.

Required fix: Decouple the #216 audit from the display enrichment: compute `_coverage_for` over the pre-enrichment `build_bullet_provenance` rows and enrich only the persisted/returned rows for the requirement chips; OR drop the `evidence_ids` union from `enrich_provenance_requirements` (the stated goal is requirement links); OR make `compute_keyword_coverage` distinguish builder evidence from claim evidence. Add a regression test: enrich a row with an unrelated claim evidence id and assert an unrelated non-corpus keyword stays `missing`.

File: docs/local-reliability-qa.md [hunk 2]
```diff
@@ -56,6 +56,7 @@ VITE_JOBHUNTER_API_BASE_URL=http://127.0.0.1:8766 pnpm web:dev -- --port 5173
...
...ared claim mappings instead of grounding them in the shipped rendered text, so the persisted "Must-have coverage 100% · passed" record can contradict the provenance-backed requirement audit on the Apply Review surface ("4/9 requirements covered" with must-haves marked missing from the tailored resume). Coverage-bearing claims must count only when bound to a shipped line (location + text binding, with the same bullet's pre-voice text honored for voice-reworded lines and the assembler's policy-gated summary swap treated as unshipped), claimed-only requirements must fail the gate with explicit shipped-resume fixes feeding the revision loop, provenance rows must carry the grounded claim requirement links so chips/badge/gate derive from one criterion, and the shipped artifact must persist a lifecycle-labeled post-voice grounded fit record (`post_generation_fit_final`) | `workers/automation/tests/test_claim_grounding.py`; `workers/automation/tests/test_requirement_led_tailoring.py` (`test_grounded_gate_reproduces_apply_review_contradiction_shape`, `test_post_generation_fit_score_rejects_claims_absent_from_shipped_text`); `workers/automation/tests/test_tailor_voice_audit_integration.py` |
```
Comment (ebarti) [inline docs/local-reliability-qa.md:59]: PR Reviewer - pr-reviewer

Severity: Medium
Gate: Pass
Evidence: This new regression-matrix row cites workers/automation/tests/test_tailor_voice_audit_integration.py as covering the grounded-gate regression, but that file is NOT modified by this PR and has no assertion on the gate `grounding` metadata, on `post_generation_fit_final` persistence/lifecycle labeling, or on the enrichment->keyword-coverage interaction (its coverage assertions use latency/python, which trace to the profile corpus and pass regardless of grounding). The only new coverage is domain-level in test_claim_grounding.py and test_requirement_led_tailoring.py.
Reproduction: Reasoned from code + grep (no grounding / post_generation_fit_final / enrich assertions in the cited integration file).

The use_cases wiring is untested end-to-end: nothing proves `_post_generation_fit_gate` now emits record["grounding"], that `execute` persists a lifecycle-labeled `post_generation_fit_final` (post_voice_shipped) with gate thresholds + residual warnings when voice degrades grounded coverage, or that a voice-reworded bullet stays grounded via prior_lines through the real pipeline. CLAUDE.md asks for a fixture that reproduces the bad state from canonical data at the owning layer, not only in the pure scorer.

Required fix: Add a use_cases-level integration test (extend test_tailor_voice_audit_integration.py) asserting the persisted `post_generation_fit_final` shape/lifecycle and the gate record `grounding` metadata; or correct the docs row to cite only tests that actually exercise the behavior.

File: workers/automation/src/jobhunter/domain/materials/use_cases.py [hunk 3]
```diff
@@ -2223,10 +2252,28 @@ def _run_candidate(
                 )
             else:
                 record["claim_mapping_validation"] = {"passed": True, "errors": []}
+            # Render this candidate's shipped lines (pure, assembler-mirroring) so
+            # the fit gate measures coverage against what would actually ship.
+            shipped_rows: tuple[BulletProvenance, ...] = ()
+            if not claim_mapping_errors:
```
Comment (ebarti) [inline workers/automation/src/jobhunter/domain/materials/use_cases.py:2258]: PR Reviewer - pr-reviewer

Severity: Low
Gate: Pass
Evidence: `shipped_rows` is only built when `not claim_mapping_errors` (use_cases.py:2258). `claim_mapping_errors` includes coverage-graph FK errors and mandatory-achievement errors, which can be non-empty while parse/binding errors are empty. In that case `_post_generation_fit_gate` does NOT early-return (its guard checks only parse + `_claim_mapping_binding_errors`, use_cases.py:643-645) and grounds mappings against shipped_rows=(), so every coverage-bearing claim is ungrounded and the gate record reports 0% grounded must-have coverage.
Reproduction: Reasoned from code.

The candidate already fails validation, so this is not a functional regression, but it attaches a misleading `post_generation_fit` record (0% coverage, "does not appear in the shipped resume" fixes) to a candidate whose real defect is a coverage-graph FK error, plus redundant duplicate gate errors -- muddying the audit trail this PR is meant to make trustworthy.

Required fix: When `claim_mapping_errors` is non-empty (or shipped_rows could not be built), skip the fit gate (return None) instead of grounding against empty rows, so the gate record reflects only structurally-sound candidates.

File: workers/automation/src/jobhunter/domain/materials/claim_grounding.py (hunk 1 above)
Comment (ebarti) [reply workers/automation/src/jobhunter/domain/materials/claim_grounding.py]: Verified fixed in f714fd44. `enrich_provenance_requirements` no longer touches `evidence_ids` (only unions requirement links); `evidence_ids_for_bullet` removed; docstring documents the #216 vector. Re-ran my repro against the new code: the enriched row's evidence_ids stay `()` and "kubernetes" remains `missing` (no longer flips to covered). #216 invariant restored. Resolving.

File: docs/local-reliability-qa.md (hunk 2 above)
Comment (ebarti) [reply docs/local-reliability-qa.md:59]: Verified fixed in f714fd44. New `test_gate_grounding_and_shipped_fit_persist_with_lifecycle_labels` in test_tailor_voice_audit_integration.py asserts the gate record carries grounding.basis + claimed_only, execute persists post_generation_fit_final (lifecycle post_voice_shipped, passed=True via the pre-voice binding after voice rewording), provenance carries the grounded requirement link, and no claim evidence leaked. Docs citation is now accurate. Full suite 1667 passed. Resolving.

File: workers/automation/src/jobhunter/domain/materials/use_cases.py (hunk 3 above)
Comment (ebarti) [reply workers/automation/src/jobhunter/domain/materials/use_cases.py:2258]: Verified fixed in f714fd44. `_run_candidate` now calls the gate only when `shipped_rows` is truthy; empty shipped rows (claim-mapping/binding or ProvenanceBindingError paths) skip the gate, so no misleading 0% record is attached to an already-rejected candidate. Resolving.

**Review by ebarti — COMMENTED** [review]
PR Reviewer - pr-reviewer: re-review of f714fd44.

Gate: PASS
Counts: Blocker=0, High=0, Medium=0, Low=0, Dismissed=0, Disagreements=0

All three findings from the prior review are correctly fixed and verified:
- HIGH (keyword-coverage #216 leak): `enrich_provenance_requirements` no longer unions `evidence_ids` (requirement links only); `evidence_ids_for_bullet` removed; docstring documents the vector. Re-ran the original repro against the new code -- enriched evidence_ids stay `()` and a non-corpus keyword stays `missing` instead of flipping to `covered`.
- MEDIUM (missing use_cases-level tests): new `test_gate_grounding_and_shipped_fit_persist_with_lifecycle_labels` asserts the gate `grounding` metadata, `post_generation_fit_final` (lifecycle `post_voice_shipped`, passed via the pre-voice binding after voice rewording), the persisted grounded requirement link, and the no-evidence-leak invariant. Docs citation now accurate.
- LOW (empty-rows gate record): the gate is skipped when `shipped_rows` is empty, so no misleading 0% is recorded on already-rejected candidates.

Verification: `pytest -q` -> 1667 passed; `ruff check .` -> clean. Three inline threads replied to and resolved.

Residual risk (unchanged, scoped to PR2): the displayed gate value (`post_generation_fit`, pre-voice) can still diverge from post-voice provenance chips in a voice-degradation case until `post_generation_fit_final` is wired to the read side.
